### PR TITLE
Improve cluster ergonomics

### PR DESCRIPTION
**Pull request by paulgb**
This fixes a few things I don't like about clusters right now:

- Since keys are (by default) shared across all clusters managed by a container, there is a weird thing where you can request a connection to a certain cluster's endpoint and get back a result from another cluster. Logically, it makes sense to consider the cluster as part of the (optional) spawn configuration. In fact, a connect request that only wants to connect to an existing backend should not need to specify a cluster at all.

- I expect the majority of self-hosting users will want only a single cluster, but they are forced to specify a cluster in every CLI command and connect request. This PR allows a `default_cluster` to be set in the controller, which is then applied if no cluster is provided in a connect request.

- Internally, even though the `backend_id` is unique across all clusters managed by the same controller, we effectively use the entire (cluster, backend) pair as a key for backends. This creates some unnecessary complexity that we can eliminate by using backend alone as the ID.

This also bumps the version to 0.4.1.

### CODE REVIEW

File: plane/src/controller/mod.rs [hunk 1]
```diff
@@ -150,14 +154,14 @@ impl ControllerServer {
...
             .route("/c/:cluster/d/:drone/drain", post(handle_drain))
             .route(
-                "/c/:cluster/b/:backend/soft-terminate",
+                "/b/:backend/soft-terminate",
                 post(terminate::handle_soft_terminate),
             )
             .route(
-                "/c/:cluster/b/:backend/hard-terminate",
+                "/b/:backend/hard-terminate",
                 post(terminate::handle_hard_terminate),
```
Comment (paulgb) [inline plane/src/controller/mod.rs:165]: This is one of the more impactful changes of this PR; a bunch of paths no longer refer to clusters.

File: plane/plane-tests/tests/backend_actions.rs [hunk 2]
```diff
@@ -6,17 +6,18 @@ use plane::{
...
 use std::time::Duration;
 
 mod common;
 
 /// Return a dummy connect request, which does not use a key.
-fn connect_request() -> ConnectRequest {
+fn connect_request(cluster: &ClusterName) -> ConnectRequest {
     ConnectRequest {
         spawn_config: Some(SpawnConfig {
+            cluster: Some(cluster.clone()),
```
Comment (paulgb) [inline plane/plane-tests/tests/backend_actions.rs:20]: Spawn config now takes an optional cluster name. If it is not provided, we fall back on the default cluster name set on the controller.

File: plane/src/types.rs [hunk 3]
```diff
@@ -285,6 +285,9 @@ impl ExecutorConfig {
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
 pub struct SpawnConfig {
+    /// Cluster to spawn to. Uses the controller default if not provided.
+    pub cluster: Option<ClusterName>,
```
Comment (paulgb) [inline plane/src/types.rs:289]: Spawn config now takes an optional cluster name. If it is not provided, we fall back on the default cluster name set on the controller.

File: plane/src/controller/backend_status.rs [hunk 4]
```diff
@@ -37,15 +36,15 @@ async fn backend_status(
...
     State(controller): State<Controller>,
 ) -> Result<Json<TimestampedBackendStatus>, Response> {
-    let status = backend_status(&controller, &cluster, &backend_id).await?;
+    let status = backend_status(&controller, &backend_id).await?;
     Ok(Json(status))
 }
 
 pub async fn handle_backend_status_stream(
-    Path((_cluster, backend_id)): Path<(ClusterName, BackendName)>, // TODO: check cluster id
+    Path(backend_id): Path<BackendName>, // TODO: check cluster id
```
Comment (rolyatmax) [inline plane/src/controller/backend_status.rs:47]: Is this comment out of date now?

Comment (paulgb) [reply plane/src/controller/backend_status.rs]: good catch.

**Review by rolyatmax — APPROVED** [review]
LGTM! I like stripping out the cluster for endpoints where it's not really required

**Pull request merged**